### PR TITLE
Implemented checkstyle with primary checks

### LINKED ISSUE #242: Build must fail when warnings arise
When testing yields warnings, we should fail the build.

Build should fail on linter issues in Java code. Ryan mentioned that the following maven plugin can help with this:
```    
  <plugin>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.6.0</version>
    <executions>
      <execution>
        <id>compile</id>
        <phase>process-sources</phase>
        <goals>
          <goal>compile</goal>
        </goals>
        <configuration>
          <compilerArgument>-Xlint:-processing</compilerArgument>
          <failOnWarning>true</failOnWarning>
        </configuration>
      </execution>
    </executions>
  </plugin>
```

### PR DESCRIPTION
**Pull request by kev-ms**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- Added stylecheck as maven plugin
- Configured style check to cause build failure in case of warnings/issues
- Suppressed some checks from standard `google_checks.xml` by adding `checkstyle-suppresions.xml`
- Fixed minor formatting issues observed.

## Observability + Testing
- What changes or considerations did you make in relation to observability?
   NA
- Did you add testing to account for any new or changed work?
   NA

## Review notes
- Some checks are surpassed to avoid bigger changes in codebase. They can be removed upon team agreement on coding standards. 

## Issues Closed or Referenced

- Closes #242

### CODE REVIEW

File: checkstyle-suppresions.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.0//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_0.dtd">
+
+<suppressions>
+  <suppress checks="Javadoc" files="."/>
+  <suppress checks="Indentation" files="."/>
+  <suppress checks="LineLength" files="."/>
```
Comment (ryancrawcour) [inline checkstyle-suppresions.xml:10]: should we suppress this? can we not override with an agree line length instead?

File: checkstyle-suppresions.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.0//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_0.dtd">
+
+<suppressions>
+  <suppress checks="Javadoc" files="."/>
+  <suppress checks="Indentation" files="."/>
```
Comment (ryancrawcour) [inline checkstyle-suppresions.xml:9]: we should agree as a team what our indentation is going to be and override the default

**Review by ryancrawcour — CHANGES_REQUESTED** [review]

File: checkstyle-suppresions.xml (hunk 2 above)
Comment (sivamu) [reply checkstyle-suppresions.xml:9]: I like the 4 space tabs, maybe we can set it as that.

File: checkstyle-suppresions.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  <suppress checks="LineLength" files="."/>
+  <suppress checks="AvoidStarImport" files="."/>
+  <suppress checks="AbbreviationAsWordInName" files="."/>
+  <suppress checks="LambdaParameterName" files="."/>
+  <suppress checks="VariableDeclarationUsageDistance" files="."/>
+  <suppress checks="CustomImportOrder" files="."/>
+</suppressions>
```
Comment (sivamu) [inline checkstyle-suppresions.xml:16]: newline needed

**Review by sivamu — COMMENTED** [review]
Does the suppressions.xml file work with this VSCode extension?
https://marketplace.visualstudio.com/items?itemName=shengchen.vscode-checkstyle

If so, we don't need sonarlint in the devcontainer json file here:
https://github.com/microsoft/kafka-connect-cosmosdb/blob/dev/.devcontainer/devcontainer.json#L46

File: checkstyle-suppresions.xml (hunk 2 above)
Comment (kev-ms) [reply checkstyle-suppresions.xml:9]: We can do custom checks but there isn't quite a way to leverage default checks and override some rules. 
We could create our own checkstyle xml to leverage both but that would isolate us from future enhancements.

Comment (ryancrawcour) [reply checkstyle-suppresions.xml]: that's not ideal. but ok. 2 spaces it is then 😂

**Review by ryancrawcour — APPROVED** [review]

**Pull request merged**